### PR TITLE
chore(wallet): migrate to Jupiter API v6

### PR DESCRIPTION
**Pull request by onyb**
Resolves https://github.com/brave/brave-browser/issues/35091

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Jupiter swaps should work as expected. Note that we don't have multiple quote options in v6 anymore, so we don't surface that it in the UI either.

### CODE REVIEW

**Review by StephenHeaps — APPROVED** [review]
iOS++

File: components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts [hunk 1]
```diff
@@ -207,19 +205,15 @@ export const useSwap = () => {
...
       }
 
-      setSelectedQuoteOptionIndex(index)
+      if (!toToken) {
+        return
+      }
+
+      setToAmount(option.toAmount.format(6))
     },
     [quoteOptions, selectedNetwork?.coin, jupiter, toToken, zeroEx.quote]
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts:218]: Can remove `selectedNetwork?.coin`, `jupiter` and `zeroEx.quote`

**Review by Douglashdaniel — APPROVED** [review]
++

File: components/brave_wallet/browser/swap_response_parser.cc [hunk 2]
```diff
@@ -234,136 +216,103 @@ mojom::ZeroExErrorPtr ParseZeroExErrorResponse(const base::Value& json_value) {
...
+  swap_quote->out_amount = quote_value->out_amount;
+  swap_quote->other_amount_threshold = quote_value->other_amount_threshold;
+  swap_quote->swap_mode = quote_value->swap_mode;
+  swap_quote->slippage_bps = quote_value->slippage_bps;
+  swap_quote->price_impact_pct = quote_value->price_impact_pct;
+
+  if (!quote_value->platform_fee.is_none()) {
+    if (!quote_value->platform_fee.is_dict()) {
       return nullptr;
     }
```
Comment (supermassive) [inline components/brave_wallet/browser/swap_response_parser.cc:291]: nit: single `if (!quote_value->platform_fee.is_dict())` would be enough

**Review by supermassive — APPROVED** [review]
wallet core lgtm

**Comment by srirambv** [comment]
Verification passed on 


Brave | 1.63.101 Chromium: 120.0.6099.199 (Official Build) nightly (64-bit)
-- | --
Revision | 8d3f0a563e185b6a763cc8e210ab24775dfb9047
OS | Linux

- Verified steps from PR
- Verified multiple quote options are removed 
- Verified Swap work as expected
- Encountered intermittent issue [#35196](https://github.com/brave/brave-browser/issues/35196)

https://github.com/brave/brave-core/assets/17010094/1f0a46a9-440d-4dd4-90e3-b72dbe572382

**Pull request merged**